pyanaconda/anaconda_logging.py

```diff
@@ -235,19 +239,35 @@ def forwardToJournal(self, logr, log_formatter=None, log_filter=None):
             journal_handler.setFormatter(log_formatter)
         logr.addHandler(journal_handler)
 
-    def stderrToJournal(self):
-        """Print all stderr messages from Anaconda to journal instead.
+    def redirect_glib_logging_to_journal(self):
+        """Redirect GLib based library logging to Journal.
+
+        Some GLib based libraries (such as GTK) do direct their
+        sometimes quite verbose log messages to the output of the
+        proces in which they are running. In the Anaconda case,
+        this creates issues with TTY1 being spammed with those
+        messages, with important content (such RDP connection intructions)
+        being scrolled out of view.
 
-        Redirect Anaconda main process stderr to Journal, as otherwise this could end up writing
-        all over the TUI on TTY1.
+        :param log: anaconda log handler
         """
+
         if not self.write_to_journal:
             return
 
-        # create an appropriately named Journal writing stream
-        anaconda_stderr_stream = journal.stream("anaconda", priority=journal.LOG_ERR)
-        # redirect stderr of this process to the stream
-        os.dup2(anaconda_stderr_stream.fileno(), sys.stderr.fileno())
+        # create functions that convert the messages comming
+        # from GLib into something that fits to the PYthon logging format
+        def log_adapter(domain, level, message, user_data):
+            self.anaconda_logger.debug("GLib: %s", message)
+
+        def structured_log_adapter(level, fields, field_count, user_data):
+            message = GLib.log_writer_format_fields(level, fields, True)
+            self.anaconda_logger.debug("GLib: %s", message)
+            return GLib.LogWriterOutput.HANDLED
+
+        # redirect GLib loug output vit the functions
+        GLib.log_set_handler(None, GLib.LogLevelFlags.LEVEL_MASK, log_adapter, None)
```

Comment:
i wonder if it would be clearer to give the individual log levels instead of using the mask wholesale. doesn't really matter, but food for thought.

Reply:
I'm more happy about this solution because:
- it is more error prone
- it will handle everything even if something would change (not expecting that but Anaconda is here a long time already and remembers a lot :D )

Reply:
I guess I see your point now.

You mean to avoid putting all the logs to `debug` anaconda log but instead set the level appropriately. That sounds like something we definitely want at least for warning and error level that would be helpful.